pkg/sqlmodel/causality.go

```diff
@@ -110,7 +110,7 @@ func genKeyString(
 		return "" // all values are `null`.
 	}
 	buf.WriteString(table)
-	return buf.String()
+	return strings.ToLower(buf.String())
```

Comment:
in line 102, we can check `columns[i].FieldType.GetCollate()` and `ToLower` the value when needed.

Reply:
updated